modules/fbs-core/web/src/app/dialogs/new-checker-dialog/new-checker-dialog.component.ts

```diff
@@ -58,10 +62,46 @@ export class NewCheckerDialogComponent implements OnInit {
       );
       this.checkerForm.controls["ord"].setValue(this.checker.ord);
     }
-    this.courseId = this.data.courseId;
-    this.taskId = this.data.taskId;
-    this.setDefaultValues();
-    this.choosedSQLChecker = false;
+
+    if (this.checker.mainFileUploaded == true) {
+      this.checkerService
+        .getChecker(this.courseId, this.taskId)
+        .pipe(map((checkers) => checkers[0]))
+        .subscribe((checker) => {
+          this.checkerService
+            .fetchMainFile(this.courseId, this.taskId, checker.id)
+            .subscribe((mainFileBlob) => {
+              this.mainFile[0] = new File([mainFileBlob], "mainFile");
+              this.fileCounter++;
+            });
+        });
+    }
+
+    if (this.checker.secondaryFileUploaded == true) {
+      this.checkerService
+        .getChecker(this.courseId, this.taskId)
+        .pipe(map((checkers) => checkers[0]))
+        .subscribe((checker) => {
+          this.checkerService
+            .fetchSecondaryFile(this.courseId, this.taskId, checker.id)
+            .subscribe((secondaryFileBlob) => {
+              this.secondaryFile[0] = new File(
+                [secondaryFileBlob],
+                "secondaryFile"
+              );
+              this.fileCounter++;
+            });
+        });
+    }
```

Comment:
These two if conditions have the same content except for the method call. I would suggest instead of the `fetchMainFile` and the `fetchSecondaryFile` methods to write a `fetchFile` method with parameter, thereby the code duplicates of the methods `fetchMainFile`, `fetchSecondaryFile`, `getMainFile`, `getSecondaryFile` would be omitted and it would be more generic for the future. In addition, the If statement  would then only affect the parameter and also the code duplication here would be omitted.

Reply:
refactoring of the fetchFile/getFile method